packages/server/src/api/controllers/query/index.ts

```diff
@@ -155,51 +231,16 @@ export async function preview(ctx: UserCtx) {
     const nestedSchemaFields: {
       [key: string]: Record<string, string | QuerySchema>
     } = {}
-    const makeQuerySchema = (
-      type: FieldType,
-      name: string,
-      subtype?: string
-    ): QuerySchema => ({
-      type,
-      name,
-      subtype,
-    })
     if (rows?.length > 0) {
       for (let key of [...new Set(keys)] as string[]) {
```

Comment:
I know you didn't write it but I think this is another case of wasteful spread operator usage. I think the following would work fine:

```javascript
for (let key of new Set(keys)) {
  // ...
}
```

And it would save converting from a `Set` to an array for no good reason.